src/stats.cpp

```diff
@@ -373,14 +342,7 @@ stats_flush_to_file(const Config& config,
     return;
   }
 
-  bool should_flush = false;
-  for (int i = 0; i < STATS_END; ++i) {
-    if (updates->data[i] > 0) {
-      should_flush = true;
-      break;
-    }
-  }
-  if (!should_flush) {
+  if (updates.all_zero()) {
```

Comment:
This is redundant since the same check is made first in the function. I'll remove it.